lib/driver.js

```diff
@@ -830,7 +830,10 @@ class XCUITestDriver extends BaseDriver {
 
     try {
       // download if necessary
-      this.opts.app = await this.helpers.configureApp(this.opts.app, '.app');
+      this.opts.app = await this.helpers.configureApp(
+        this.opts.app,
+        this.opts.realDevice ? '.ipa' : '.app'
```

Comment:
I assume we should allow both extension. At the end we anyway only can use .app, because both simulator and real device installers only support this package type. I configureApp must be switched to the new bundles processing strategy similarly it has been done in https://github.com/appium/appium-espresso-driver/pull/750 and https://github.com/appium/appium-espresso-driver/pull/758/files

Reply:
@mykola-mokhnach You can better assess why this change is needed with this companion PR in base driver https://github.com/appium/appium-base-driver/pull/556

Reply:
we don't push anything into the base driver anymore. It has been moved to the Appium 2 monorepo https://github.com/appium/appium/tree/2.0/packages/base-driver and is now maintained there. Basically, the idea is this code must not be located in base driver at all, because it's strictly iOS-specific. 

Also, I like the idea with a single .zip package, although it would make more sense to simply analyze the source .app bundle inside it and select the first bundle that matches the target architecture rather than just the very first one. Check https://github.com/appium/appium-xcuitest-driver/blob/a94c2023228c0bff8c6645cf2612a61b0a31c7b2/lib/app-utils.js#L40 on how to verify the matching platform.

We cannot assume that .ipa packages only target real devices or vice versa, because .ipa is still a vanilla .zip archive with the same .app bundle in it.

Reply:
> we don't push anything into the base driver anymore. It has been moved to the Appium 2 monorepo 

Thanks for the heads up

> The idea is this code must not be located in base driver at all, because it's strictly iOS-specific.

I understand this should not be in Base driver but then we have a problem.

Base driver is deciding what file to use in the zip but it's not allowed to know about how to filter the files...
[Currently we are just grabbing the first item](https://github.com/appium/appium/blob/b359f29df5187f0b4e3200309e5f3fa02b061dea/packages/base-driver/lib/basedriver/helpers.js#L508-L509) in the zip that meets supported extensions.

How do you think we can proceed with this feature?

  - Give each driver an `unzipApp` that knows how to do it and remove the base driver unzipApp
  - Introduce dependency in base driver, refactor unzipApp to be passed a function that validates if a single item in a zip file is or not supported instead of a list supported extensions, then we could do something like
    ```js
    import { verifyApplicationPlatform } from 'xcuitest-driver/somewhere'
    
    opts.isIOS && unzipApp(archive, verifyApplicationPlatform)
    ```

Reply:
Have you actually checked the actual code from PRs I mentioned above?
In the current base driver revision the configureApp method has been [extended](https://github.com/appium/appium/blob/b359f29df5187f0b4e3200309e5f3fa02b061dea/packages/base-driver/lib/basedriver/helpers.js#L158) and made more flexible to be able to handle multiple package types.

Reply:
Ah yes, indeed hadn't seen the PR as I was short on time, thanks.
I'll see what I can do with the post processing extension.